uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -12,6 +16,28 @@ function tasksassignments_civicrm_config(&$config) {
 }
 
 /**
+ * @param array $wrappers
+ * @param array $apiRequest
+ */
+function tasksassignments_civicrm_apiWrappers(&$wrappers, $apiRequest) {
+  $requestJson = ArrayHelper::value('json', $_REQUEST);
+  $params = ArrayHelper::value('params', $apiRequest, []);
+  $optionGroup = ArrayHelper::value('option_group_id', $params);
+  $isActivityType = $optionGroup === 'activity_type';
+  $entity = ArrayHelper::value('entity', $apiRequest);
+  $action = ArrayHelper::value('action', $apiRequest);
+
+  $jsonFromCaseEdit = '"actTypes":["OptionValue","get",{"option_group_id":"activity_type","options":{"sort":"name","limit":0}}]';
+  $isAssignmentEdit = strpos($requestJson, $jsonFromCaseEdit) !== FALSE;
+
+  if ($isAssignmentEdit && $entity === 'OptionValue' && $action === 'get' && $isActivityType) {
+    $wrappers[] = new StripInvalidTaskTypesWrapper();
+  } elseif ($entity === 'CaseType' && $action === 'create') {
```

Comment:
I am of course not happy with frontend validation only , I actually do not like the way both T&A and hrrequirment depends on civicrm case tables  and many other things in other extensions and that's why I think why we should stop everything we do in civihr and sit to do some better planning because for me the whole thing is a mess and without writing some plans sooner that later nothing is going to improve.

but since I don't have the decision here I would like at least not breaking existing sutff ( which will leads to more hacks in this case) , and since there is a high rate for this modification to break thing so  I would like to avoid it .